src/test_utils/katana/genesis.rs

```diff
@@ -0,0 +1,319 @@
+    ) -> Result<Self> {
+        let private_key: Option<B256> = private_key.into();
+        let evm_address: Option<Address> = evm_address.into();
+
+        let evm_address = evm_address
+            .and_then(|addr| FieldElement::from_byte_slice_be(addr.as_slice()).ok())
+            .or(private_key.and_then(|pk| self.evm_address(pk).ok()));
+        let evm_address = evm_address.ok_or(eyre!("Failed to derive EVM address"))?;
+
+        let kakarot_address = self.cache_load("kakarot_address")?;
+        let eoa_class_hash = self.eoa_class_hash()?;
+        let proxy_class_hash = self.proxy_class_hash()?;
+
+        // Set the eoa storage
+        let eoa_storage = [
+            (storage_addr("evm_address")?, evm_address),
+            (storage_addr("kakarot_address")?, kakarot_address),
+            (storage_addr("_implementation")?, eoa_class_hash),
+        ]
+        .into_iter()
+        .collect::<HashMap<_, _>>();
+
+        let eoa =
+            GenesisContractJson { class: proxy_class_hash, balance: None, nonce: None, storage: Some(eoa_storage) };
+
+        let starknet_address = self.compute_starknet_address(evm_address)?;
+        self.contracts.insert(starknet_address, eoa);
+
+        // Set the allowance for the EOA to the Kakarot contract.
+        let key = get_storage_var_address("ERC20_allowances", &[*starknet_address, kakarot_address])?;
+        let storage =
+            [(key, FieldElement::from(u128::MAX)), (key + 1u8.into(), FieldElement::from(u128::MAX))].into_iter();
+        self.token_storage.extend(storage);
+
+        // Write the address to the Kakarot evm to starknet mapping
+        let kakarot_address = ContractAddress::new(kakarot_address);
+        let kakarot_contract = self.contracts.get_mut(&kakarot_address).ok_or(eyre!("Kakarot contract missing"))?;
+        kakarot_contract
+            .storage
+            .get_or_insert_with(HashMap::new)
+            .extend([(get_storage_var_address("evm_to_starknet_address", &[evm_address])?, starknet_address.0)]);
+
+        Ok(self)
+    }
+
+    /// Fund the starknet address deployed for the evm address of the passed private key
+    /// with the given amount of tokens.
+    pub fn fund(mut self, pk: B256, amount: U256) -> Result<Self> {
+        let evm_address = self.evm_address(pk)?;
+        let starknet_address = self.compute_starknet_address(evm_address)?;
+        let eoa = self.contracts.get_mut(&starknet_address).ok_or(eyre!("Missing EOA contract"))?;
+
+        let key = get_storage_var_address("ERC20_balances", &[*starknet_address])?;
+        let low = amount & U256::from(u128::MAX);
+        let low: u128 = low.try_into().unwrap(); // safe to unwrap
+        let high = amount >> U256::from(128);
+        let high: u128 = high.try_into().unwrap(); // safe to unwrap
+
+        let storage = [(key, FieldElement::from(low)), (key + 1u8.into(), FieldElement::from(high))].into_iter();
+        self.token_storage.extend(storage);
```

Comment:
what if this account already has a balance?

Reply:
I'm not sure what you mean, how could it have a balance?

Reply:
if you call fund twice on the same acct

Reply:
one will overwrite the other during loading of the genesis.